### ISSUE TITLE
Bluetooth autoplay

### ISSUE THREAD

**Issue by seijikun** [OxygenCobalt/Auxio#235]
#### Describe the feature you want to implement:
A functionality I miss from many players is that the player auto-**spawns** when the smartphone connects to a bluetooth audio devices.
I'm explicitly talking about auto-spawn, meaning that the player is started automatically. Not that it starts playback when it is already running anyway.
For example:
- I enter my car, the car radio turns on its bluetooth receiver, my smartphone connects to it -> player starts playing automatically.
- I plan to go for a jog, I turn on my bluetooth headphones, my smartphone connects to them -> player starts playing automatically.

#### Do other music players handle this? If so, how?
Yes, there are some. The most prominent is [PowerAmp](https://powerampapp.com/).
[Pulse](https://f-droid.org/packages/com.hardcodecoder.pulsemusic/) also handled this, before it became closed-source. (Couldn't find the functionality in the source archive - so he probably removed it there)

#### Why do you think this will improve everyone's usage of Auxio?
I think a bluetooth audio device that should auto-start playing as soon as its in range - like a car - is a pretty common scenario.
Another example is bluetooth headphones, that are basically only ever active (= connected) if you also intend to use them (i.e. listen to music).
For these use-cases its an awesome usability improvement!

#### Due Diligence:
- [x] I have read the [Contribution Guidelines](https://github.com/OxygenCobalt/Auxio/blob/dev/.github/CONTRIBUTING.md).
- [x] I have read the [Accepted Additions and Requests](https://github.com/OxygenCobalt/Auxio/blob/dev/info/ADDITIONS.md) document.
- [x] I have checked for this feature in the [FAQ](https://github.com/OxygenCobalt/Auxio/blob/dev/info/FAQ.md).

**Comment by OxygenCobalt** [comment]
Pretty good work,  a few things though:

First, some UI things:
- Say I connect my earbuds to my phone. Will this cause it to play exactly at the moment that the audio stream starts routing to my Earbuds? If this is the case, then I will gladly accept this feature. I've wanted to add this, but I'm only aware of the ACM broadcasts which didn't quite work to my desires
- 
 The service start functionality is slightly wrong. BroadcastRecievers are 
3. I would prefer something like this to be unified with the existing "Headset Autoplay" feature.

Then, some issues I've noticed with the code:
- Avoid `Log.e(tag, "<function name>")` in favor of `logD("Recieved bluetooth A2DP request)`. This is just personal convention.
- I don't want to request permissions on startup. I would rather bind the permission check to the setting itself. I'll only query for permission when the user toggles the setting on. If the permission is revoked, the value will be set to false again.
- The service initialization is quite wonky. `RestoreState` is already started when the activity is created, and since broadcast receivers are only active when the app is alive, that call doesn't work. Plus, `RestoreState` can take ~300ms or more, which could cause the app to crash as the service won't enter foreground in time. Wouldn't it be more correct to just start playing?

**Comment by etyarews** [comment]
Just throwing this out here: If possible, I wish this feature to work with wired headsets. I don't use Bluetooth, and the reason I asked for #244 was basically just to have more control over autoplay

**Comment by OxygenCobalt** [comment]
Sadly I've already rejected allowing this functionality to also be able to initialize the app for the same reasons that #244 was rejected @etyarews. It is just so hard to initialize the app without explicitly opening the app, at least with the current framework.

**Comment by Donkey-Doug** [comment]
Option to disable this functionality if it gets added please! Not a fan of autoplay.

**Comment by is-code-afish** [comment]
Speaking of power amp, you can get a similar effect using the equalizer.

**Comment by OxygenCobalt** [comment]
What do you mean exactly @is-code-afish?

**Comment by osnafreak** [comment]
This is a great idea, the Automatic playing in the car is very usefull for me too. The best way is to make it optional.

**Comment by leandroprz** [comment]
> This is a great idea, the Automatic playing in the car is very usefull for me too. The best way is to make it optional.

I agree. An option to enable or disable this would be the best way to implement it.
